app/serializers/song_serializer.rb

```diff
@@ -1,5 +1,11 @@
 class SongSerializer < ActiveModel::Serializer
-  attributes :id, :title, :album, :duration
+  attributes :id, :title, :album, :duration, :full_title
+
   embed :ids
+
   has_one :artist
+
+  def duration
+    object.duration_in_minutes_and_seconds
```

Comment:
See above. This is where you'd access the `decorated_song` instead of `object` which in this case is the wrapped decorated model from the controller.